### PR TITLE
[PostgreSQL] Advanced Sync rules

### PR DESCRIPTION
**Pull request by moxarth-elastic**
## Closes https://github.com/elastic/connectors-py/issues/1690

This PR introduces advanced filtering rules for PostgreSQL by queries to filter out document based on that
Added documentation with several examples illustrating the syntax, how a basic use case works
Included tooltip on the configurable fields that are ignored when advanced sync rules are used

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes

## Release Note

For existing PostgreSQL connectors, the next sync after the upgrade will delete all the existing documents and reindex them with different document IDs.

### CODE REVIEW

**Comment by moxarth-elastic** [comment]
buildkite test this

File: connectors/sources/postgresql.py [hunk 1]
```diff
@@ -371,54 +481,126 @@ async def fetch_documents(self, table):
...
+        except (InternalClientError, ProgrammingError) as exception:
+            self._logger.warning(
+                f"Something went wrong while fetching document for table {table}. Error: {exception}"
+            )
+
+    async def fetch_documents_from_query(self, tables, query):
+        """Fetches all the data from the given query and format them in Elasticsearch documents
+
+        Args:
+            table (str): Name of table
```
Comment (wangch079) [inline connectors/sources/postgresql.py:496]:
```suggestion
            tables (str): List of tables
```

**Review by wangch079 — DISMISSED** [review]
LGTM, only a small nit

**Review by wangch079 — APPROVED** [review]

**Comment by parth-crest** [comment]
buildkite test this

**Pull request merged**